### PR TITLE
[New Rule] Unsigned DLL Side-Loading from a Suspicious Folder

### PR DESCRIPTION
**Pull request by Samirbous**
New endpoint rule to detect potential DLL sideload from suspicious directories : 

image

Looks for DLL loaded from one of the listed dir and unsigned and loaded by a signed process and from the same process.executable current directory.

### CODE REVIEW

File: rules/windows/defense_evasion_unsigned_dll_loaded_from_suspdir.toml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                  "?:\\Windows\\media\\*.dll",
+                  "?:\\Windows\\Globalization\\*.dll",
+                  "?:\\Windows\\L2Schemas\\*.dll",
+                  "?:\\Windows\\LiveKernelReports\\*.dll",
+                  "?:\\Windows\\ModemLogs\\*.dll",
+                  "?:\\Windows\\ImmersiveControlPanel\\*.dll",
+                  "?:\\$Recycle.Bin\\*.dll") and 
+	 
+	 /* DLL loaded from the process.executable current directory */
+	 endswith~(substring(dll.path, 0, length(dll.path) - (length(dll.name) + 1)), substring(process.executable, 0, length(process.executable) - (length(process.name) + 1)))
```
Comment (terrancedejesus) [inline rules/windows/defense_evasion_unsigned_dll_loaded_from_suspdir.toml:114]: 🧙🏽

**Review by terrancedejesus — APPROVED** [review]

File: rules/windows/defense_evasion_unsigned_dll_loaded_from_suspdir.toml (hunk 1 above)
Comment (Samirbous) [reply rules/windows/defense_evasion_unsigned_dll_loaded_from_suspdir.toml:114]: correct that's the intent of the query logic. 👍

**Review by w0rk3r — APPROVED** [review]
Next level wizardry here, LGTM 🔥

File: rules/windows/defense_evasion_unsigned_dll_loaded_from_suspdir.toml [hunk 2]
```diff
@@ -0,0 +1,135 @@
+[metadata]
+creation_date = "2022/11/22"
+maturity = "production"
```
Comment (w0rk3r) [inline rules/windows/defense_evasion_unsigned_dll_loaded_from_suspdir.toml:3]:
```suggestion
maturity = "production"
integration = ["endpoint"]
```

**Pull request merged**